ark/util/serialize.ts

```diff
@@ -47,7 +47,10 @@ export type snapshot<t, depth extends 1[] = []> =
 	: depth["length"] extends 10 ? unknown
 	: t extends array<infer item> ? array<snapshot<item, [...depth, 1]>>
 	: {
-			[k in keyof t]: snapshot<t[k], [...depth, 1]>
+			[k in keyof t as k extends symbol ? string : k]: snapshot<
```

Comment:
Can we extract out the logic from `snapshotPrimitive` and use that here instead? That way symbolic keys don't get widened to `string`.

Reply:
updated! 

I changed `snapshotPrimitive` though, so I think it would be good if you double check that that is as intended. The tests still pass and the parentheses match the actual use now, so I think it's correct after the change.